### PR TITLE
SONAR-6229 Indexing of activity logs consumes too much memory during server startup

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-6229 Indexing of activity logs consumes too much memory during server startup

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java [hunk 1]
```diff
@@ -19,46 +19,33 @@
...
 import org.sonar.core.activity.db.ActivityDto;
 import org.sonar.core.activity.db.ActivityMapper;
+import org.sonar.core.persistence.DaoComponent;
 import org.sonar.core.persistence.DbSession;
-import org.sonar.server.db.BaseDao;
-import org.sonar.server.search.IndexDefinition;
+import org.sonar.core.persistence.MyBatis;
 
-import java.util.List;
+public class ActivityDao implements DaoComponent, ServerComponent {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java]: why both DaoComponent and ServerComponent are needed ?

File: server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java [hunk 2]
```diff
@@ -19,46 +19,33 @@
...
+  public ActivityDao(MyBatis mybatis) {
+    this.mybatis = mybatis;
   }
 
-  @Override
-  protected ActivityDto doInsert(DbSession session, ActivityDto item) {
-    mapper(session).insert(item);
-    return item;
+  public void insert(ActivityDto dto) {
+    DbSession session = mybatis.openSession(false);
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java]: don't you want to manage the createdAt field at the dao layer ?

File: server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityDoc.java [hunk 3]
```diff
@@ -19,54 +19,88 @@
...
+  @VisibleForTesting
+  ActivityDoc() {
+    super(new HashMap<String, Object>());
+  }
+
+  public void setCreatedAt(Date date) {
+    setField(ActivityIndexDefinition.FIELD_CREATED_AT, date);
+  }
+
+  public Date createdAt() {
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityDoc.java]: getCreatedAt to be compliant with naming in this class

File: server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityIndex.java [hunk 4]
```diff
@@ -19,119 +19,86 @@
...
   }
 
-  public SearchResponse search(ActivityQuery query, QueryContext options,
-    @Nullable FilterBuilder domainFilter) {
-
-    // Prepare query
-    SearchRequestBuilder esSearch = getClient()
-      .prepareSearch(this.getIndexName())
-      .setTypes(this.getIndexType())
-      .setIndices(this.getIndexName());
```
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityIndex.java]: don't you need to set indices ?

**Comment by teryk** [comment]
LGTM, I have a good understanding of what you've done, but I couldn't do a review as thorough as desired

File: server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java (hunk 1 above)
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java]: Currently DaoComponent do not extend BatchComponent nor ServerComponent, as some of them are still used by batch. Target goal is that all DAOs are ServerComponent.

File: server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityIndex.java (hunk 4 above)
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/activity/index/ActivityIndex.java]: yes, through paramter of prepareSearch() ! That was a duplication.

**Comment by simonbrandhof** [comment]
Feedback applied. Thanks

**Pull request closed without merging**